docs/llm_tools.md

```diff
@@ -0,0 +1,204 @@
+| `name`              | string     | The name of the tool. Required.                                                                                |
+| `description`       | string     | Description of the tool to help the LLM understand when and how it should be called. Optional but recommended. |
+| `parameters`        | vol.Schema | The voluptuous schema of the parameters. Defaults to vol.Schema({})                                            |
+
+The `llm.Tool` class has the following user-defined methods:
+
+#### `async_call`
+Perform the actual operation of the tool when called by the LLM. This must be an async method and return json-serializable result. Its arguments are `hass` and an instance of `llm.ToolInput`.
+
+Example:
+```python
+class MyTool(llm.Tool):
+    ...
+    async def async_call(self, hass: HomeAssistant, tool_input: llm.ToolInput) -> Any:
+        """Call the tool."""
+        return {"result": "success"}
+```
+
+The `ToolInput` has following attributes:
+
+| Name              | Type    | Description                                                                                             |
+|-------------------|---------|---------------------------------------------------------------------------------------------------------|
+| `tool_name`       | string  | The name of the tool being called                                                                       |
+| `tool_args`       | dict    | The arguments provided by the LLM. The arguments are converted and validated using `parameters` schema. |
+| `platform`        | string  | The DOMAIN of the conversation agent using the tool                                                     |
+| `context`         | Context | The `homeassistant.core.Context` of the conversation                                                    |
+| `user_prompt`     | string  | The raw text input that initiated the tool call                                                         |
+| `language`        | string  | The language of the conversation agent, or "*" for any language                                         |
+| `agent_id`        | string  | The ID of the conversation agent                                                                        |
+| `conversation_id` | string  | The ID of the conversation                                                                              |
+| `device_id`       | string  | The device ID that user uses for the conversation, if available                                         |
+| `assistant`       | string  | The assistant name used to control exposed entities. Currently only `conversation` is supported.        |
+
+### Unregistration
+When your integration no longer wants to provide the tool, you need to unregister it with `llm.async_remove_tool`.
+
+Example:
+```python
+async_remove_tool(hass, MyTool)
+```
+
+You can also use the tool name:
+```python
+async_remove_tool(hass, "multiply")
+```
+
+### Full example
```

Comment:
There is already effectively a full example above for `MyTool` so I don't think both are needed and one should be removed or combined.

Reply:
You are right, this example used to be more extensive before I removed `custom_serializer`.